torch/_functorch/aot_autograd.py

```diff
@@ -981,6 +1194,9 @@ def inner(*flat_args):
             num_intermediate_bases=len(intermediate_bases),
             keep_input_mutations=keep_input_mutations,
             traced_tangents=traced_tangents,
+            subclass_inp_meta=create_subclass_meta(flat_args),
+            subclass_out_meta=create_subclass_meta(fw_graph_outs),
```

Comment:
suggestoin: name this more clearly than just "out" because there are multiple notions of output here

Reply:
Agreed, this is a pain:

The full forward graph outputs consist of `(*mutated_inps, *user_outs, *intermediate_bases, *saved_for_bw_tensors)`

The first 3 of those 4 can be subclasses (but not saved_for_bw_tensors), and so `fw_graph_outs` here corresponds to info about the subclass-ness of all of those fw outputs (including indexing nformation about their global output order)